Tests/MongoSwiftSyncTests/MongoCollectionTests.swift

```diff
@@ -589,6 +589,78 @@ final class MongoCollectionTests: MongoSwiftTestCase {
             .to(throwError(errorType: MongoError.InvalidArgumentError.self))
     }
 
+    func testFindAndModifyWithHint() throws {
+        try withTestNamespace { client, _, collection in
+            let requirements = TestRequirement(
+                minServerVersion: ServerVersion(major: 4, minor: 4)
```

Comment:
According to the [spec change](https://github.com/mongodb/specifications/commit/0f803725487cc951877f6975061dd1e24056a045#diff-2fe314a3da48fca10d3a170b09a5a393a93dbb4e9369d6741c99656a95dd7034R1609), drivers are actually in charge of returning errors if the server is < 4.2. Could we add a separate test case that verifies we return such an error?

Reply:
done